docs/cartesian-product.adoc

```diff
@@ -0,0 +1,222 @@
+:page-title: Testing all test parameter combinations (Cartesian product)
+:page-description: Extends JUnit Jupiter with `@CartesianProductTest`, a special `@ParameterizedTest` which tests all combinations of its input
+
+From Wikipedia, the free encyclopedia:
```

Comment:
"the free encyclopedia" seems a bit patronizing, I think most people know that small project already. :wink: